src/hypothesis/extra/datetime.py

```diff
@@ -123,3 +123,26 @@ def times(allow_naive=None, timezones=None):
 
 def datetime_to_time(dt):
     return dt.timetz()
+
+SECS_IN_DAY = 3600 * 24
+MICROS_IN_SEC = 1000000
+MICROS_IN_DAY = MICROS_IN_SEC * SECS_IN_DAY
+class TimedeltaStrategy(SearchStrategy):
+
+    def __init__(self, min_value=dt.timedelta.min, max_value=dt.timedelta.max):
+        self.max_micros = max_value.microseconds + (max_value.seconds + max_value.days * SECS_IN_DAY) * MICROS_IN_SEC
+        self.min_micros = min_value.microseconds + (min_value.seconds + min_value.days * SECS_IN_DAY) * MICROS_IN_SEC
```

Comment:
There’s nothing (directly) checking that these values make sense – in particular, that `min_value < max_value`. You’ll get an `AssertionError` that bubbles out of `cu.integer_range()`, but it would be much better to provide a user-friendly error message here, in particular one that includes the values supplied by the user.